src/linear_algebra/tensor_product.lean

```diff
@@ -475,6 +475,19 @@ begin
 end
 end
 
+variables (R M N)
+
+/-- The simple (aka pure) elements span the tensor product. -/
+lemma span_tmul_eq_top :
+  submodule.span R { t : M ⊗[R] N | ∃ m n, m ⊗ₜ n = t } = ⊤ :=
+begin
```

Comment:
I wonder if it makes sense to have a `set.range2` or `function.surjective2` to describe this type of thing? Out of scope for this PR at any rate.

Reply:
Thanks for the quick review!

Out of curiosity what do you have in mind for `set.range₂`, `function.surjective₂`?

Reply:
Oh I think I maybe understand. Do you mean something like:
```lean
def set.range₂ (f : α → β → γ) := set.range (function.uncurry f)
```
etc.?

Reply:
That's probably more frustrating to work with than `def set.range₂ (f : α → β → γ) := { t : γ | ∃ m n, f m n = t }`

Note we already have `function.injective2`, but I suspect it's not really used.

Reply:
Thanks, I think I understand now.